iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/analyzer/Analysis.java

```diff
@@ -751,7 +752,7 @@ public TsBlock constructResultForMemorySource(MPPQueryContext context) {
 
   @Override
   public boolean isQuery() {
-    return false;
+    return getStatement().isQuery();
```

Comment:
record whether it is query in analysis while analyzing, delete isQuery method in Statement